frontend/src/components/common/Resource/Resource.tsx

```diff
@@ -563,6 +566,246 @@ export function ConditionsTable(props: ConditionsTableProps) {
+        value = error.message;
+      }
+
+      variables.set(name, {
+        value,
+        from,
+        isError: !!error,
+        isSecret: false,
+      });
+    };
+
+    // Function to process all keys from a secret and add them to the variables map
+    const processAllSecretKeys = (secretName: string, prefix: string) => {
+      const [secret, setSecret] = React.useState<Secret | null>(null);
+      const [error, setError] = React.useState<ApiError | null>(null);
+      ResourceClasses.Secret.useApiGet(setSecret, secretName, props.namespace, setError);
+
+      if (error) {
+        variables.set(`${prefix}${secretName}`, {
+          from: `secret/${secretName}`,
+          value: error.message,
+          isError: true,
+          isSecret: false,
+        });
+      } else if (secret?.data) {
+        Object.entries(secret.data).forEach(([key, value]) => {
+          variables.set(`${prefix}${key}`, {
+            from: `secret/${secretName}`,
+            value: atob(value),
+            isError: false,
+            isSecret: true,
+          });
+        });
+      }
+    };
+
+    // Function to process all keys from a configMap and add them to the variables map
+    const processAllConfigMapKeys = (configMapName: string, prefix: string) => {
+      const [configMap, setConfigMap] = React.useState<ConfigMap | null>(null);
+      const [error, setError] = React.useState<ApiError | null>(null);
+      ResourceClasses.ConfigMap.useApiGet(setConfigMap, configMapName, props.namespace, setError);
+
+      if (error) {
+        variables.set(`${prefix}${configMapName}`, {
+          from: `configMap/${configMapName}`,
+          value: error.message,
+          isError: true,
+          isSecret: false,
+        });
+      } else if (configMap?.data) {
+        Object.entries(configMap.data).forEach(([key, value]) => {
+          variables.set(`${prefix}${key}`, {
+            from: `configMap/${configMapName}`,
+            value,
+            isError: false,
+            isSecret: false,
+          });
+        });
+      }
+    };
+
+    // Process env variables directly from env
+    props?.env?.forEach(item => {
+      if (item.value) {
+        variables.set(item.name, {
+          value: item.value,
+          from: 'manifest',
+          isSecret: false,
+          isError: false,
+        });
+      } else if (item.valueFrom) {
+        const ref = item.valueFrom;
+        if (ref.secretKeyRef) {
+          processValueFromSecret(item.name, ref.secretKeyRef);
+        } else if (ref.configMapKeyRef) {
+          processValueFromConfigMap(item.name, ref.configMapKeyRef);
+        }
+      }
+    });
+
+    // Process env variables from envFrom (configMap or secret references)
+    props?.envFrom?.forEach(item => {
+      if (item.secretRef) {
+        processAllSecretKeys(item.secretRef.name, prefix);
+      } else if (item.configMapRef) {
+        processAllConfigMapKeys(item.configMapRef.name, prefix);
+      }
+    });
```

Comment:
I am not sure this logic will not end up violating hooks' rules. Maybe @sniok can help review it.

Reply:
You can only call hooks inside components or inside other hooks, these processX functions should be components: processValueFromSecret, processValueFromConfigMap, processAllSecretKeys, processAllConfigMapKeys, processValueFromFieldRef, processValueFromResourceFieldRef.